persist/sqlite/consensus_refactored_test.go

```diff
@@ -0,0 +1,943 @@
+		genesisBlock.Transactions[0].SiacoinOutputs[0].Address = addr1
+	})
+
+	unlockConditions := types.StandardUnlockConditions(pk1.PublicKey())
+	scID := genesisBlock.Transactions[0].SiacoinOutputID(0)
+	txn1 := types.Transaction{
+		SiacoinInputs: []types.SiacoinInput{{
+			ParentID:         scID,
+			UnlockConditions: unlockConditions,
+		}},
+		SiacoinOutputs: []types.SiacoinOutput{{
+			Address: addr2,
+			Value:   genesisBlock.Transactions[0].SiacoinOutputs[0].Value,
+		}},
+	}
+	testutil.SignTransaction(cs, pk1, &txn1)
+
+	prevState := cs
+	b := testutil.MineBlock(cs, []types.Transaction{txn1}, types.VoidAddress)
+	cs, au := applyUpdate(t, cs, store.SupplementTipBlock(b), b)
+	if err := db.UpdateChainState(nil, []chain.ApplyUpdate{au}); err != nil {
+		t.Fatal(err)
+	}
+
+	{
+		sces, err := db.SiacoinElements([]types.SiacoinOutputID{scID})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(sces)", 1, len(sces))
+
+		sce := sces[0]
+		testutil.Equal(t, "sce.Source", explorer.SourceTransaction, sce.Source)
+		testutil.Equal(t, "sce.SpentIndex", cs.Index, *sce.SpentIndex)
+		testutil.Equal(t, "sce.SiacoinElement.SiacoinOutput", genesisBlock.Transactions[0].SiacoinOutputs[0], sce.SiacoinOutput)
+	}
+
+	{
+		sces, err := db.SiacoinElements([]types.SiacoinOutputID{txn1.SiacoinOutputID(0)})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(sces)", 1, len(sces))
+
+		sce := sces[0]
+		testutil.Equal(t, "sce.Source", explorer.SourceTransaction, sce.Source)
+		testutil.Equal(t, "sce.SpentIndex", nil, sce.SpentIndex)
+		testutil.Equal(t, "sce.SiacoinElement.SiacoinOutput", txn1.SiacoinOutputs[0], sce.SiacoinOutput)
+	}
+
+	{
+		txns, err := db.Transactions([]types.TransactionID{txn1.ID()})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(txns)", 1, len(txns))
+
+		testutil.CheckTransaction(t, txn1, txns[0])
+	}
+
+	ru := revertUpdate(t, prevState, store.SupplementTipBlock(b), b)
+	if err := db.UpdateChainState([]chain.RevertUpdate{ru}, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	{
+		sces, err := db.SiacoinElements([]types.SiacoinOutputID{scID})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(sces)", 1, len(sces))
+
+		sce := sces[0]
+		testutil.Equal(t, "sce.Source", explorer.SourceTransaction, sce.Source)
+		testutil.Equal(t, "sce.SpentIndex", nil, sce.SpentIndex)
+		testutil.Equal(t, "sce.SiacoinElement.SiacoinOutput", genesisBlock.Transactions[0].SiacoinOutputs[0], sce.SiacoinOutput)
+	}
+
+	{
+		sces, err := db.SiacoinElements([]types.SiacoinOutputID{txn1.SiacoinOutputID(0)})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(sces)", 0, len(sces))
+	}
+
+	{
+		txns, err := db.Transactions([]types.TransactionID{txn1.ID()})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(txns)", 0, len(txns))
+	}
+}
+
+func TestEphemeralSiacoinOutput(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	addr1 := types.StandardUnlockHash(pk1.PublicKey())
```

Comment:
(nit) since you need the unlock conditions anyway I'd create them up here as well and go from private key -> unlock conditions -> unlock hash to avoid passing the public key twice

```go
pk1 := types.GeneratePrivateKey()
uc1 := types.StandardUnlockConditions(pk1.PublicKey())
addr1 := uc1.UnlockHash()
```
you can also avoid `addr1` and just inline the `uc1.UnlockHash` call.